Library/Homebrew/cli/named_args.rb

```diff
@@ -324,7 +324,11 @@ def resolve_latest_keg(name)
         # Return keg if it is the only installed keg
         return kegs if kegs.length == 1
 
-        kegs.reject { |k| k.version.head? }.max_by(&:version)
+        stable_kegs = kegs.reject { |k| k.version.head? }
+
+        return kegs.max_by { |keg| Tab.for_keg(keg).source_modified_time } if stable_kegs.blank?
```

Comment:
I included the `source_modified_time` code here directly since it is so brief. If you think this should be encapsulated elsewhere for use in `resolve_latest_keg` and `latest_head_version`, I'm happy to make changes. @Rylan12

Reply:
@cnnrmnn I think this is fine. I'm not sure it really makes much sense to extract what's essentially `#source_modified_time` to its own method.

Just making sure, the [`revision` secondary sort from `Formula#latest_head_version`](https://github.com/Homebrew/brew/blob/5ea678bbc7307f476b0f01ac727099300db95f83/Library/Homebrew/formula.rb#L556) isn't necessary here, right?

Reply:
I'm not sure that the `revision` secondary sort is needed anywhere we're dealing exclusively with `HEAD` versions. If you check out the [parsing function for `PkgVersion`](https://github.com/Homebrew/brew/blob/5ea678bbc7307f476b0f01ac727099300db95f83/Library/Homebrew/pkg_version.rb#L20), you'll see that it parses the revision using a regex. `HEAD` paths (e.g., `HEAD-6974ce8`) matched to that regex will always set `revision` to nil. Any `HEAD` version should have a revision equal to `0` since `to_i` is called on the parsed `nil` `revision`.

Reply:
Gotcha makes sense. So it could probably be removed from `latest_head_version` too, right? Probably not worth doing but just curious if there's a difference.

Reply:
> So it could probably be removed from latest_head_version too, right?

Yep, should be fine to remove. I can do that in a quick PR as to not confuse anyone dealing with that code in the future.

Reply:
You definitely don't need to but it probably wouldn't hurt. Plus it will expose more maintainers to it in case anyone knows something we're missing and it turns out it _is_ necessary

Reply:
@Rylan12 Did some quick digging, and found [a spec](https://github.com/Homebrew/brew/blob/cd900ef71eeecb621faba88aabc641fe93325fe6/Library/Homebrew/test/formula_spec.rb#L347) that tests `HEAD` versions with revisions. Do you know if there's any scenario where a `HEAD` version would actually have a revision?

I would think so given that the above tests exists, but I'm still curious. If you don't know, I can do a little more digging and restore the secondary sort here if appropriate.

Reply:
Ah, good call. If you do `brew install --HEAD` on a formula that's specified a `revision` it _will_ show up as `HEAD-6974ce8_3` (which means there will be a revision), so I think it _does_ need to be added here.

Reply:
Gotcha, will push a fix shortly.